packages/manager/modules/vps/src/dashboard/tile/configuration/service.js

```diff
@@ -0,0 +1,164 @@
+      upperPlanTotalPriceValue - currentPlanTotalPriceValue,
+      head(currentPlan.prices).price,
+    );
+  }
+
+  static getPlanPriceValue(plan) {
+    return plan.prices.reduce((total, { price }) => total + price.value, 0);
+  }
+
+  static getPriceStructure(value, { currencyCode, text }) {
+    return {
+      currencyCode,
+      text: text.replace(/\d+(?:[.,]\d+)?/, value.toFixed(2)),
+      value,
+    };
+  }
+
+  /**
+   *  Set the vps informations
+   *
+   *  @param {Object} vps      The informations provided by 2API
+   *  @param {Object} vpsModel The model attribute provided by
+   *                           GET /vps/{servicName} (not present in 2API...)
+   *
+   *  @return {ConfigurationTileService}
+   */
+  setVps(vps, vpsModel) {
+    this.vps = vps;
+    this.vpsModel = vpsModel;
+
+    return this;
+  }
+
+  /**
+   *  Set available upgrades in order to get the current vps plan.
+   *
+   *  @param  {Array}   availableUpgrades response from the GET /order/upgrade/vps/{serviceName} API call.
+   *
+   *  @return {ConfigurationTileService}
+   */
+  setAvailableUpgrades(availableUpgrades) {
+    this.availableUpgrades = availableUpgrades;
+
+    return this;
+  }
+
+  /**
+   *  Get informations about the upgrades available from configuration tile in the dashboard.
+   *  These plans will be useful for upgrade modal.
+   *
+   *  @param  {Object}  catalog           response from the GET /order/catalog/public/virtualprivateserver.
+   *
+   *  @return {Object}
+   */
+  getAvailableUpgrades(catalog) {
+    if (this.vpsModel.vcore === 8) {
+      // if vps elite - no upgrade available from configuration tile
+      return {
+        memory: {
+          plan: null,
+        },
+        storage: {
+          plan: null,
+        },
+      };
+    }
+
+    // get next ram plan infos
+    const nextRamVps = find(catalog.products, ({ blobs }) => {
+      if (!get(blobs, 'technical')) {
+        return false;
+      }
+
+      const cpu = get(blobs, 'technical.cpu.cores');
+      const storage = get(blobs, 'technical.storage.disks[0].capacity');
+      const ram = get(blobs, 'technical.memory.size');
+
+      return (
+        cpu === this.vps.vcore &&
+        storage === this.vpsModel.disk &&
+        ram === this.vps.ram.value * 2
+      );
+    });
+
+    const nextRamVpsPlan = nextRamVps
+      ? find(this.availableUpgrades, {
+          planCode: nextRamVps.name,
+        })
+      : null;
+
+    // get next storage plan infos
+    const nextStorageVps = find(catalog.products, ({ blobs }) => {
+      if (!get(blobs, 'technical')) {
+        return false;
+      }
+
+      const cpu = get(blobs, 'technical.cpu.cores');
+      const ram = get(blobs, 'technical.memory.size');
+      const storage = get(blobs, 'technical.storage.disks[0].capacity');
+
+      return (
+        cpu === this.vps.vcore &&
+        ram === this.vps.ram.value &&
+        storage === this.vpsModel.disk * 2
```

Comment:
The `* 2` looks like magic to me. Maybe a comment on which rule is applied could help to understand the logic here

Reply:
Ok. I will add it